backend/src/main/java/fk/prof/backend/model/assignment/impl/SimultaneousWorkAssignmentCounterImpl.java

```diff
@@ -0,0 +1,36 @@
+package fk.prof.backend.model.assignment.impl;
+
+import fk.prof.backend.model.assignment.SimultaneousWorkAssignmentCounter;
+import fk.prof.backend.proto.BackendDTO;
+import recording.Recorder;
+
+/**
+ * No operation of this class is thread-safe
+ * Right now, only invoked from backend daemon thread, so we are good
+ */
+public class SimultaneousWorkAssignmentCounterImpl implements SimultaneousWorkAssignmentCounter {
+  private final int maxAllowedConcurrentWorkAssignmentSlots;
+  private int currentlyAcquiredWorkAssignmentSlots = 0;
+
+  public SimultaneousWorkAssignmentCounterImpl(int maxAllowedConcurrentWorkAssignmentSlots) {
+    this.maxAllowedConcurrentWorkAssignmentSlots = maxAllowedConcurrentWorkAssignmentSlots;
+  }
+
+  /**
+   * TODO: Should be refactored to take process group and work prototype as input and accordingly determine slots
+   * @return
+   */
+  public int acquireSlots(Recorder.ProcessGroup processGroup, BackendDTO.WorkProfile workProfile) {
```

Comment:
Well, for this method to determine that, it'll have to do the same calculation as planner, won't it?

Planner is already doing it, it can ask capacity in terms of score etc, this class can be abstract, it just says it has capacity or it doesn't etc.